lexicon/providers/yandexcloud.py

```diff
@@ -0,0 +1,307 @@
+"""Module provider for Yandex.Cloud DNS
+
+API doc: https://cloud.yandex.com/en/docs/dns/
+
+If you are using the service account, dns.viewer required for viewing the DNS records
+and dns.editor for editing them.
+
+If Cloud ID and Folder ID are not specified,
+resource-manager.clouds.member and resource-manager.viewer are needed as well.
+"""
+import json
+import logging
+import urllib
+from typing import List, Dict
+
+import requests
+
+from lexicon.exceptions import AuthenticationError
+from lexicon.providers.base import Provider as BaseProvider
+
+__author__ = "Dmitry Verkhoturov"
+__license__ = "MIT"
+__contact__ = "https://github.com/paskal"
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ["yandexcloud.net"]
+
+
+def provider_parser(subparser):
+    """Generate parser provider for Yandex"""
+    subparser.add_argument(
+        "--auth-token",
+        required=True,
```

Comment:
Every cli flag can also be set with environment variables, so none of them must be required. However it is totally fine to error out in the code afterward if `auth_token` has not been set by any available configuration approach.